packages/api-core/src/createApolloServer.js

```diff
@@ -84,6 +92,10 @@ export default function createApolloServer(options = {}) {
     subscriptions = {
       path
     };
+
+    const wsServer = new WebSocketServer({ server: httpServer, path: DEFAULT_GRAPHQL_PATH });
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    wsServerCleanup = useServer({ schema, context: { ...contextFromOptions } }, wsServer);
```

Comment:
I believe you need to grab the `context(session)` function being passed to `new ApolloServer`, define it above here instead, and pass in that function rather than `context: { ...contextFromOptions }`. In that function there is `if (connection) return context;`, but you might need to make it smarter than that. In general, confirm that auth checks work, or document that subscriptions cannot require auth. Also, it might make sense to move the `contextFuncs` loop to before the `if (connection) return context;` (or generally to make sure it runs for WS).

See also the "ws server usage with custom context value" section [here](https://github.com/enisdenjo/graphql-ws)

Reply:
I noticed after reviewing this that some of the context changes I propose are already done in https://github.com/reactioncommerce/reaction/pull/6671/files. So just pull in those changes and add some tests to confirm.